visualization/app/codeCharta/ui/slider/slider.component.spec.ts

```diff
@@ -1,56 +1,56 @@
-import { TestBed } from "@angular/core/testing"
-import { render, screen } from "@testing-library/angular"
-import userEvent from "@testing-library/user-event"
-import { drag } from "../../util/testUtils/drag"
+import { HarnessLoader } from "@angular/cdk/testing"
+import { TestbedHarnessEnvironment } from "@angular/cdk/testing/testbed"
+import { ComponentFixture, TestBed } from "@angular/core/testing"
+import { MatInputHarness } from "@angular/material/input/testing"
+import { MatSliderHarness } from "@angular/material/slider/testing"
 import { SliderComponent } from "./slider.component"
 import { SliderModule } from "./slider.module"
 
+let loader: HarnessLoader
+let fixture: ComponentFixture<SliderComponent>
+
 describe("SliderComponent", () => {
 	beforeEach(() => {
 		TestBed.configureTestingModule({
-			imports: [SliderModule]
-		})
+			imports: [SliderModule],
+			declarations: [SliderComponent]
+		}).compileComponents()
+		fixture = TestBed.createComponent(SliderComponent)
+		loader = TestbedHarnessEnvironment.loader(fixture)
```

Comment:
do we really need to dig into Angular's internals? I will check, if I can get the tests working again with testing-library style.

Reply:
That is actually the recommended way in the Material UI docu (with harnesses). Actually the reason why a lot of tests broke (and why this took so long to migrate) is that we used to access internal Material UI classes and hardcode them. Then Material decided to change almost all internal classes, and some parts of the API. To avoid this from happening again, I started using their recommended way with the Test harnesses they provide. Which will be updated when material updates. So no extra work on our part.
I only see upsides to this, so can you shed some light why this is bad?

Reply:
Those tests were also super annoying to fix as they were before, but as soon as I switched to using the harnesses it went in a few minutes and was more readable and concise.
Here is more to it here:
https://material.angular.io/guide/using-component-harnesses

Reply:
Awesome! Glad to hear that we can use both 😄